File: examples/load_annotation.R
# Quick-start: load the coral annotation table in R.
library(tidyverse)

# Load annotation table (protein_sequence excluded from GitHub-hosted file)
ann <- read_tsv("species/Gfas/annotation/Gfas_master_annotation.tsv")
cat("Loaded:", nrow(ann), "genes x", ncol(ann), "columns\n")

# Annotation status breakdown
ann %>% count(annotation_status)

# Confidence tier breakdown
ann %>% count(confidence_tier)

# Gold-tier secreted ligands
ligands <- ann %>% filter(confidence_tier == "Gold", cellchat_role == "Ligand")
cat("Gold-tier ligands:", nrow(ligands), "\n")

# Parse pipe-delimited GO terms into long format
go_long <- ann %>%
  select(gfas_gene_id, all_go_terms) %>%
  separate_rows(all_go_terms, sep = "\\|") %>%
  filter(!is.na(all_go_terms))

# Join to Seurat metadata (example)
# seurat_meta <- FetchData(seurat_obj, vars = c("gene_id", "seurat_clusters"))
# merged <- seurat_meta %>% left_join(ann, by = c("gene_id" = "gfas_gene_id"))

File: examples/load_annotation.py
"""
Quick-start: load the coral annotation table in Python.
"""
import pandas as pd

# Load annotation table (protein_sequence excluded from GitHub-hosted file)
df = pd.read_csv("species/Gfas/annotation/Gfas_master_annotation.tsv", sep="\t")
print(f"Loaded: {df.shape[0]} genes x {df.shape[1]} columns")

# Annotation status breakdown
print("\nAnnotation status:")
print(df["annotation_status"].value_counts())

# Confidence tier breakdown
print("\nConfidence tiers:")
print(df["confidence_tier"].value_counts())

# Gold-tier secreted ligands
ligands = df[(df["confidence_tier"] == "Gold") & (df["cellchat_role"] == "Ligand")]
print(f"\nGold-tier ligands: {len(ligands)}")

# Parse pipe-delimited GO terms into a list
df["go_list"] = df["all_go_terms"].str.split("|")

# Join to Seurat metadata (example)
# seurat_meta = pd.read_csv("seurat_metadata.csv", index_col=0)
# merged = seurat_meta.join(df.set_index("{species}_gene_id"), on="gene_id")  # replace {species}
